raft/raft.go

```diff
@@ -654,6 +671,11 @@ func stepLeader(r *raft, m pb.Message) {
 					// an update before, send it now.
 					r.sendAppend(m.From)
 				}
+				// Transfer leadership is in progress.
+				if m.From == r.leadTransferee && pr.Match == r.raftLog.lastIndex() && !r.timeoutNowSent {
+					r.logger.Infof("%x sent MsgTimeoutNow to %x after received MsgAppResp", r.id, m.From)
```

Comment:
@bdarnell, @xiang90, I feel we cannot send `MsgTimeoutNow` as soon as we have sent a ` MsgApp`  containing the last index in all cases, if we add that code and remove the code to send `MsgTimeoutNow ` when a `MsgAppResp` is received, the test case `TestLeaderTransferToSlowFollower` cannot be passed as node 3 will reject that `MsgApp` just before the `MsgTimeoutNow` triggered by `pb.MsgTransferLeader` as its log is not up-to-date, but the following `MsgTimeoutNow` will trigger node3 to start an election which also will fail.

1. Maybe one way is we keep the code which send `MsgTimeoutNow` as receiving `MsgAppResp`, but add one optimistic sending `MsgTimeoutNow` only when we optimistically update the `pr.Next`, but this may also have the risk that peer may reject that `MsgApp` and  `MsgTimeoutNow` will make the group unavailable for one election timeout. And also we have to keep the `timeoutNowSent` flag.

2. We keep the current code as we can also remove the `timeoutNowSent` flag. This is the updated commit.